tests/unit/stacked-vertical-bar-test.js

```diff
@@ -0,0 +1,792 @@
+// and more thorough testing of slice sorting mechanics.
+var sliceSortingData = _.cloneDeep(three_ranges).concat([{
+  sliceLabel: "Label 4",
+  barLabel: "Group One",
+  value: -10
+}, {
+  sliceLabel: "Label 4",
+  barLabel: "Group Two",
+  value: -10
+}, {
+  sliceLabel: "Label 4",
+  barLabel: "Group Three",
+  value: 15
+}, {
+  sliceLabel: "Label 5",
+  barLabel: "Group One",
+  value: 10
+}, {
+  sliceLabel: "Label 5",
+  barLabel: "Group Three",
+  value: 10
+}, {
+  sliceLabel: "Label 6",
+  barLabel: "Group One",
+  value: 20
+}, {
+  sliceLabel: "Label 6",
+  barLabel: "Group Three",
+  value: 20
+}]);
+
+moduleForComponent('stacked-vertical-bar-chart', '[Unit] Stacked bar component', {
+  needs: [ 'template:components/chart-component'],
+  beforeEach: function () {},
+  afterEach: function () {}
+});
+
+test('Margins are the right size', function(assert) {
+  assert.expect(3);
+  var component = this.subject();
+
+  assert.equal(component.get('marginLeft'), 0, 'no left margin');
+  assert.equal(component.get('marginRight'), 0, 'no right margin');
+  assert.equal(component.get('marginBottom'), 30, 'has top margin (because it has a legend)');
+});
+
+test('In total, the correct number of stacking slices is displayed', function(assert) {
+  assert.expect(1);
+  this.subject({data: three_ranges});
+  this.render();
+
+  assert.equal(this.$('svg rect').length, three_ranges.length,
+    "For " + three_ranges.length + " data points, that many stacking slices are shown");
+});
+
+// Passed in to Array.prototype.sort(), which has the bogus default behavior of
+// coercing every value in the array to a string and sorting alphabetically
+const diff = function(x, y) {
+  return (x - y);
+};
+
+const getFloatAttrValueOfSvgSlice = function(slice, attributeName) {
+  return parseFloat(slice.attributes[attributeName].value);
+};
+
+const getHeightOfSvgSlice = function(slice) {
+  return getFloatAttrValueOfSvgSlice(slice, 'height');
+};
+
+/**
+ * Helper function to extract transform values from an HTML element.
+ * These values are stored inside a string in the 'transform' attribute, so
+ * some massaging must be done.
+ *
+ * For example, if the HTML looks as follows:
+ *   <g class='tick' transform='translate(0.0, 42.0)' ... />
+ * then calling this helper with this element will return either 0.0 or 42.0,
+ * depending on whether index is 0 or 1.
+ * @function
+ * @param  {jQuery} $element - HTML element wrapped in jQuery
+ * @param  {number} index - Either 0 or 1, whether to extract the x transform
+ *                          value or y transform value
+ * @return {float}
+ */
+const getFloatTransformValue = function($element, index) {
+  return parseFloat($element.attr('transform').match(/\d+(\.\d+)?/g)[index]);
+};
+
+const PERCENT_TOLERANCE = 0.0001;
+
+const equalsWithTolerance = function(actual, expected) {
+  var tolerance = Math.abs(expected * PERCENT_TOLERANCE);
+  return Math.abs(actual - expected) < tolerance;
```

Comment:
I didn't use an absolute error tolerance initially because that doesn't work well as the numbers being compared become bigger. If you have two numbers that both come out of a complex set of floating-point calculations, and one of them is `1,000,000,001.0` and the other is `1,000,000,000.0`, you can make a compelling argument that they are in fact equal once the errors in each of the calculations are fixed. But an absolute delta wouldn't accept them as equal.

With these smaller numbers (`< 1000`), though, I suppose an absolute delta is okay.